chain/vm/fvm.go

```diff
@@ -13,20 +19,209 @@ import (
 	ffi "github.com/filecoin-project/filecoin-ffi"
 	ffi_cgo "github.com/filecoin-project/filecoin-ffi/cgo"
 
+	"github.com/filecoin-project/lotus/chain/actors/adt"
+	"github.com/filecoin-project/lotus/chain/actors/builtin/miner"
 	"github.com/filecoin-project/lotus/chain/types"
 	"github.com/ipfs/go-cid"
 )
 
 var _ VMI = (*FVM)(nil)
+var _ ffi_cgo.Externs = (*FvmExtern)(nil)
 
 type FvmExtern struct {
 	Rand
 	blockstore.Blockstore
+	epoch   abi.ChainEpoch
+	lbState LookbackStateGetter
+	base    cid.Cid
+}
+
+// Similar to the one in syscalls.go used by the Lotus VM, except it never errors
+// Errors are logged and "no fault" is returned, which is functionally what go-actors does anyway
```

Comment:
It would be nice if we could return a better error. E.g., distinguish between invalid arguments and "no fault". But we can do that later.

Reply:
Yeah, sounds like a future Steb problem.